src/dev/flang/ast/AbstractFeature.java

```diff
@@ -467,15 +467,14 @@ String qualifiedName(AbstractFeature context)
   {
     var tfo = state().atLeast(State.FINDING_DECLARATIONS) && outer() != null && outer().isCotype() ? outer().cotypeOrigin() : null;
     return
-      /* special type parameter used for this.type in type features */
-      isCoTypesThisType() ? (tfo != null ? tfo.qualifiedName(context) : "null") + ".this.type (in type feature)" :
-
-      /* cotype: use original name and add ".type": */
-      isCotype()             &&
-      cotypeOrigin() != null                 ? cotypeOrigin().qualifiedName(context) + ".type"
-
-      /* a normal feature name */
-                                                  : qualifiedName0(context);
+      isCoTypesThisType()
+        /* special type parameter used for this.type in type features */
+        ? (tfo != null ? tfo.qualifiedName(context) : "null") + ".this.type"
+        : isCotype() && cotypeOrigin() != null
+        /* cotype: use original name and add ".type": */
+        ? cotypeOrigin().qualifiedName(context) + ".type"
+        /* a normal feature name */
+        : qualifiedName0(context);
```

Comment:
I find the nested ternary operators a bit confusing. I think the readability could be improved by either changing the formatting or avoiding the nesting.

```suggestion
    return
      isCoTypesThisType()
        /* special type parameter used for this.type in type features */
        ? (tfo != null ? tfo.qualifiedName(context) : "null")
          + ".this.type"
        : isCotype() && cotypeOrigin() != null
          /* cotype: use original name and add ".type": */
          ? cotypeOrigin().qualifiedName(context) + ".type"
          /* a normal feature name */
          : qualifiedName0(context);
```

```suggestion
    if (isCoTypesThisType())
      {
        /* special type parameter used for this.type in type features */
        return (tfo != null ? tfo.qualifiedName(context) : "null") + ".this.type";
      }
    else
      {
        return isCotype() && cotypeOrigin() != null
          /* cotype: use original name and add ".type": */
          ? cotypeOrigin().qualifiedName(context) + ".type"
          /* a normal feature name */
          : qualifiedName0(context);
      }
```

Reply:
@fridis does not like this style of having multiple returns. you would need to introduce a result field.